### PR TITLE
Adds a component to make any storage tactical-reload capable.

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Closes https://github.com/tgstation/TerraGov-Marine-Corps/pull/12705
Closes https://github.com/tgstation/TerraGov-Marine-Corps/pull/12692

Creates a new component that allows any storage/armor module to be tactical reload capable.
Adds the component to a couple storages/modules that I believe should be tactical reload capable.
image

## Why It's Good For The Game
Makes it easy to make it so you can right click a storage to tactical reload your gun.
## Changelog
:cl:
code: New component that allows storages to be tac reloadable.
/:cl:

### CODE REVIEW

File: code/modules/clothing/modular_armor/attachments/storage.dm
Comment (ivanmixo) [inline code/modules/clothing/modular_armor/attachments/storage.dm:22]: Why not just set this to true instead of setting it to true manually on every subtype?

Comment (Xander3359) [reply code/modules/clothing/modular_armor/attachments/storage.dm:22]: Because I want the var on storage level without having to set it to false on every single storage that disallows it.

Comment (ivanmixo) [reply code/modules/clothing/modular_armor/attachments/storage.dm:22]: I suggest setting it to be explicitly FALSE instead of null at least, then. For clarity.

**Comment by FosterGrime-Heymynameslouie** [comment]
spederngemergh instantly

**Comment by Lumipharon** [comment]
I'll tm this after I look through it, but uh, don't merge. This is a very large marine buff.

**Comment by Xsdew** [comment]
Storage Tactical Reloads are the only thing magazine fed sidearms have as a unique mechanic over Primary Arms. While a small feature it helps makes up for small magazine capacity of handguns. This is of course only if you ignore pulling out new handguns with Quick Equip.

**Comment by TiviPlus** [comment]
@Lumipharon whats going on here

**Review by TiviPlus — APPROVED** [review]
lgtm

**Pull request merged**
